docs: notes on migration to `@fluentui/react-components@9.0.0-rc.1` (#21629)
microsoft/fluentui: packages/react-components/MIGRATION-NOTES.md

```ts
import { shorthands } from '@fluentui/react-componenents';

console.log(shorthands.overflow('hidden')); // { overflowX: 'hidden', overflowY: 'hidden' }
```

Please apply following changes:

```diff
import { makeStyles } from '@fluentui/react-componenents';
+import { makeStyles, shorthands } from '@fluentui/react-componenents';

const useStyles = makeStyles({
-  backgroundColor: { background: 'red' },
+  backgroundColor: { backgroundColor: 'red' },
-  padding: { padding: '5px' },
+  padding: { ...shorthands.padding('5px') },
-  margin: { margin: '5px' },
+  margin: { ...shorthands.margin('5px') },
-  border: { border: '5px solid red' },
+  border: { ...shorthands.border('5px', 'solid', 'red') },
-  borderRight: { borderRight: '5px solid red' },
+  borderRight: { ...shorthands.borderRight('5px', 'solid', 'red') },
});
```

For more details, please check [microsoft/fluentui#20573](https://github.com/microsoft/fluentui/pull/20573).

### makeStyles is Griffel now [just rename]

`makeStyles` CSS-in-JS become a separate project called [Griffel](https://github.com/microsoft/griffel). It still used in Fluent UI React v9.

## Typings & exports

> ### Hooks are export with "\_unstable" suffix

Author: ```suggestion
### Hooks are exported with "\_unstable" suffix
```

> # Notes on migration to `@fluentui/react-components@9.0.0-rc.1`

Reviewer: can we specify migration from what version are we talking? ( based on previous convo it should be from Beta to RC)

Reviewer 2: ```suggestion
# Migration notes from `@fluentui/react-components@9.0.0-beta.5`to `@fluentui/react-components@9.0.0-rc.1`
```

```ts
import { shorthands } from '@fluentui/react-componenents';

console.log(shorthands.overflow('hidden')); // { overflowX: 'hidden', overflowY: 'hidden' }
```

Please apply following changes:

```diff
import { makeStyles } from '@fluentui/react-componenents';
+import { makeStyles, shorthands } from '@fluentui/react-componenents';

const useStyles = makeStyles({
-  backgroundColor: { background: 'red' },
+  backgroundColor: { backgroundColor: 'red' },
-  padding: { padding: '5px' },
+  padding: { ...shorthands.padding('5px') },
-  margin: { margin: '5px' },
+  margin: { ...shorthands.margin('5px') },
-  border: { border: '5px solid red' },
+  border: { ...shorthands.border('5px', 'solid', 'red') },
-  borderRight: { borderRight: '5px solid red' },
+  borderRight: { ...shorthands.borderRight('5px', 'solid', 'red') },
});
```

For more details, please check [microsoft/fluentui#20573](https://github.com/microsoft/fluentui/pull/20573).

### makeStyles is now Griffel [just rename]

`makeStyles` CSS-in-JS become a separate project called [Griffel](https://github.com/microsoft/griffel). It is still used in Fluent UI React v9.

## Changes to the theming system

### Brand ramp updated

> The brand colors have were updated to match the latest design.

Reviewer 3: ```suggestion
The brand colors have been updated to match the latest design guidelines.
```

#### New `Slot` type, and renamed slot utility types

> **Note**: This change should not affect most users of the library. It only affects authors of custom components.

All slots are now declared using a single `Slot` type:

- `IntrinsicShorthandProps` => `Slot`: direct rename with the same arguments.
- `ObjectShorthandProps` => `Slot`: the argument changes from the props type to the component type; e.g. from `ButtonProps`, to `typeof Button`.

```diff
type MyComponentSlots = {
-  root?: IntrinsicShorthandProps<'div'>;
-  slotA?: IntrinsicShorthandProps<'label', 'span' | 'div'>;
-  slotB?: ObjectShorthandProps<ButtonProps>;
+  root?: Slot<'div'>;
+  slotA?: Slot<'label', 'span' | 'div'>;
+  slotB?: Slot<typeof Button>;
};
```

The following types related to slots have been renamed:

- `ShorthandProps` => `WithSlotShorthandValue`
- `ShorthandRenderFunction` => `SlotRenderFunction`
- `ObjectShorthandPropsRecord` => `SlotPropsRecord`
- `DefaultObjectShorthandProps` => `UnknownSlotProps`

## Component changes

### `Accordion`

- `AccordionHeaderExpandIcon` has been removed and replaced by `ChevronRightRegular` from `@fluentui/react-icons`.

> - `AccordionHeader` props `children` is no longer a slot [#21285](https://github.com/microsoft/fluentui/pull/21285)

Reviewer 3: ```suggestion
- `AccordionHeader` props `children` is no longer a slot. [#21285](https://github.com/microsoft/fluentui/pull/21285)
```

> # Notes on migration to `@fluentui/react-components@9.0.0-rc.1` from `9.0.0-beta.5`

Reviewer 2: ```suggestion
# Migration notes from `@fluentui/react-components@9.0.0-beta.5`to `@fluentui/react-components@9.0.0-rc.1`
```
